openssl-dynamic/src/main/c/ssl_private.h

```diff
@@ -356,23 +356,30 @@ struct tcn_ssl_task_t {
 tcn_ssl_task_t* tcn_ssl_task_new(JNIEnv*, jobject);
 void tcn_ssl_task_free(JNIEnv*, tcn_ssl_task_t*);
 
+typedef struct tcn_ssl_state_t tcn_ssl_state_t;
+struct tcn_ssl_state_t {
+    int handshakeCount;
+    tcn_ssl_ctxt_t *ctx;
+    tcn_ssl_task_t* ssl_task;
+    tcn_ssl_verify_config_t* verify_config;
+};
+
+#define TCN_GET_SSL_CTX(ssl, C)                             \
+    TCN_BEGIN_MACRO                                         \
+        tcn_ssl_state_t* _S = tcn_SSL_get_app_state(ssl);   \
+        if (_S == NULL) {                                   \
+            C = NULL;                                       \
```

Comment:
I also dont understand this.   If _S is null, wont it still be derefenced unconditionally below?

Reply:
Good catch... missed an else :(